add feedback template page (#3076)
grommet/hpe-design-system: aries-site/src/pages/templates/user-feedback-collection.mdx

import { FeedbackAnatomy, SimpleFeedbackExample, SolicitedFeedbackExample } from '../../examples';
import { Example } from '../../layouts';
import { Checkmark } from 'grommet-icons';

export const contact = {
  name: 'Doris Singer',

>   email: 'doris.singer@hpe.com',

Reviewer: Given this is a public facing site, just confirming we are fine putting Doris's email directly here? I know we've done this with the HPE Global Header team's general email but just checking.

Reviewer 2: Good catch. I'd recommend removing and maybe placing a link to something internal that has her contact info?

Author: checking with Doris on this

};

Feedback collected from users provides qualitative insights into users' perception of ease-of-use 
and value delivered. Qualitative measures compliment quantitative usage and interaction data 
to form a well-rounded analytic picture of the customer experience.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/SimpleFeedbackExample.js"
  docs="https://v2.grommet.io/starrating?theme=hpe"

>   figma="https://www.figma.com/file/StvDVAzAJvZzzTNmd7TQVu/Feedback-components?node-id=1023%3A40808&t=mu8ZYNQhot1rkRHX-4"

Reviewer: ```suggestion
  figma="https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=5554%3A77765&t=TNCIYkwaIfFrdWxf-1"
```
This is the link to the published component. Current link was working file.

>
  <SimpleFeedbackExample />
</Example>

## When to use
When and where to collect feedback from users should be a collaborative effort working with the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a>.

> This team will work with you to construct a questionare aligned with your objectives and 

Reviewer: ```suggestion
This team will work with you to construct a questionnaire aligned with your objectives and 
```

[provide implementation details](#feedback-setup-and-submission) such as API credentials and endpoints for submitting 
feedback responses.

In-app feedback collection should be considered where it aids the understanding of users' 
perceptions around the ease-of-use and/or satisfaction with particular content, features, 
or workflows.

> Feedback may be [actively solicited](#solicited-feedback) or [collected passively](#capturing-unsolicited-feedback).

Reviewer: ```suggestion
Feedback may be [actively solicited](#solicited-feedback) or [passively collected](#capturing-unsolicited-feedback).
```
Just creating parallel structure in the anchor naming

### When not to use
Don’t collect feedback in the following use cases:
- If it disrupts a user flow or is intrusive in any way.
- There is no actionable plan for what to do with the data.
- In locations where users do not have enough context to form a valid opinion. 
- There are too many instances soliciting feedback which can [dilute feedback effectiveness](#sampling-frequency).

### Sampling frequency
Avoid over sampling and/or diluting effectiveness by limiting who is asked for feedback and 
how often. Consider implementing "rules of engagement" specifying under what conditions an 
individual user should be solicited for feedback. Example engagement rules might be:

> - Has not submitted response to this questionare in the last 90 days.

Reviewer: ```suggestion
- Has not submitted response to this questionnaire in the last 90 days.
```

> - Has not been asked to complete this questionare in the last 30 days.

Reviewer: ```suggestion
- Has not been asked to complete this questionnaire in the last 30 days.
```

| **2**  | **Body**     | The body may be populated as many questions as appropriate.          | <Checkmark a11yTitle="true" size="small" /> | <span>Question types can include [StarRating](https://v2.grommet.io/starrating), [ThumbsRating](https://v2.grommet.io/thumbsrating), select all that apply (e.g. checkbox group), choose one (e.g. RadioButtonGroup), open ended (e.g. TextArea), or any other input types.</span> |                                                             
| **3**  | **Footer **  | Contains two actions: "Submit" and "Dismiss" if used in Layer modal. Otherwise only "Submit" is needed inline. | <Checkmark a11yTitle="true" size="small" /> |  Submit button sends survey responses to the HPE customer experience team and initiates thank you message. If presented in a modal, the Dismiss button allows the user to bypass the survey and continue their product experience. |

## Feedback setup and submission

> To iniate user feedback collection, please contact the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a> for the following steps:

Reviewer: ```suggestion
To initiate user feedback collection, please contact the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a> for the following steps:
```

> To iniate user feedback collection, please contact the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a> for the following steps:

Reviewer: I'm not sure the wording of "for the following steps" makes sense to me. Will Doris's team send the following to the person who contacts? Or it's more that they will contact Doris to initiate the interaction. Then, the following back-and-forth will occur with Doris to set up the feedback

Author: so they will just work with Doris for this information.. Doris will give them everything they need for the API calls

Reviewer 3: I can take a stab at re-writing for clarity.

1. A dedicated survey will be created for your use case. At that time, an API token, survey id, and response id will be provided.
1. Obtain the survey response schema with the [retrieve a survey response call](https://api.qualtrics.com/1179a68b7183c-retrieve-a-survey-response). The properties in the "values" object represent the data which should be submitted with each new survey response.
1. Survey responses are submitted using [create a new response](https://api.qualtrics.com/f1cad92018d2b-create-a-new-response).

> ```

Reviewer: Remove? Or are we planning to show example "onSubmit" code here?

Reviewer 3: Code sample added.

const fasdf = () 
```

## Examples

### Solicited feedback

> Solicited feedback is an active method for asking users for their opinions and perspectives. 

Reviewer: ```suggestion
Solicited feedback is an active method of asking users for their opinions and perspectives. 
```

A feedback questionaire is presented after a user has completed a task or consumed a piece of 
content. Solicited feedback should not interupt or impede a user from completing their intended 
objective.

> #### Task flow in center-layer

Reviewer: ```suggestion
#### Task flow in center layer
```

Click on "Power 1 Device" to see the progressive example of the feedback questions. 
<Example
  code={[
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/PowerDeviceFeedback.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/SolicitedFeedbackExample.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/PowerOnBodyExample.js',
  ]}
  github="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/SolicitedFeedbackExample.js"
  showResponsiveControls={['fullScreen']}
  screenContainer
  height="650px"
  width="100%"
>
  <SolicitedFeedbackExample />
</Example>

> ### Capturing unsolicited feedback

Reviewer: ```suggestion
### Unsolicited feedback
```
small bit of simplification to align with "Solicited feedback". soft opinion

Author: Im okay with just `Unsolicited feedback` but I believe @halocline added the `Capturing unsolicited` so just want his feedback

Reviewer 3: Soft opinion. Either is fine with me.

Unsolicited feedback provides users the opportunity to ask questions, provide ideas, or 
make recommendations. Acting similar to a “suggestions box,” this mechanism is placed in 

> a persistent location, such as within a app-level menu or embedded with the application's

Reviewer: ```suggestion
a persistent location, such as within an app-level menu or embedded with the application's
```

import { FeedbackAnatomy, SimpleFeedbackExample, ModalTaskFlowExample } from '../../examples';
import { Example } from '../../layouts';
import { Checkmark } from 'grommet-icons';

export const contact = {

>   name: 'DS System Staff',

Reviewer 3: ```suggestion
  name: 'Unified Customer Experience team',
```

Author: ooh right forgot we were pretending for mean time

>   email: 'designsystem-staff@hpe.com',

Reviewer 3: 👍

};

Feedback collected from users provides qualitative insights into users' perception of ease-of-use 
and value delivered. Qualitative measures compliment quantitative usage and interaction data 
to form a well-rounded analytic picture of the customer experience.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/feedback/SimpleFeedbackExample.js"
  docs="https://v2.grommet.io/starrating?theme=hpe"
  figma="https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=5554%3A77765&t=TNCIYkwaIfFrdWxf-1"
>
  <SimpleFeedbackExample />
</Example>

## When to use

> When and where to collect feedback from users should be a collaborative effort working with the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a>.

Reviewer 3: ```suggestion
When and where to collect feedback from users should be a collaborative effort working with the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>Unified Customer Experience team</a>.
```

This team will work with you to construct a questionnaire aligned with your objectives and 
[provide implementation details](#feedback-setup-and-submission) such as API credentials and endpoints for submitting 
feedback responses.

In-app feedback collection should be considered where it aids the understanding of users' 
perceptions around the ease-of-use and/or satisfaction with particular content, features, 
or workflows.

Feedback may be [actively solicited](#solicited-feedback) or [passively collected](#capturing-unsolicited-feedback).

### When not to use
Don’t collect feedback in the following use cases:
- If it disrupts a user flow or is intrusive in any way.
- There is no actionable plan for what to do with the data.

> - In locations where users do not have enough context to form a valid opinion. 

Reviewer 4: Is it possible for us to elaborate a little bit more on this. Is there a couple of examples we can provide to let our users know where Feedback shouldn't be utilized? Like in the middle of a check out form? On a detailed view page? Just thinking out loud because I found myself asking when would be a few scenarios where I wouldn't be able to use Feedback.

Reviewer 4: Maybe we can include some Do's and Don't examples within this section? This can be something we do later on.

Author: sounds good I can add that to the follow on ticket we have for next steps

- There are too many instances soliciting feedback which can [dilute feedback effectiveness](#sampling-frequency).

> ### Sampling frequency

Reviewer 4: How does sampling frequency work? Is there a log I have access to that gives me an overview of how many times i'm asked/requested for feedback? It would be nice to understand this a little further as a designer. So as a designer, how do I ask my developer to ensure that we're requesting feedback every 120 days?

Reviewer 3: Sampling frequency would be something that the product and customer experience teams would need to arrive at. It is a balance between collecting enough data to make informed decisions, but not too much that it overburdens the user, negatively impacting user experience.

Perhaps we insert "Sampling frequency is a balance between collecting enough data to make informed decisions, but not too often that becomes a burden, negatively impacting user experience. Product and customer experience teams should work together to arrive at a well-balanced frequency." in between the first sentence and "Consider ..."?

Reviewer 4: Thank you for explaining, I think that sounds good!

Avoid over sampling and/or diluting effectiveness by limiting who is asked for feedback and 
how often. Consider implementing "rules of engagement" specifying under what conditions an 
individual user should be solicited for feedback. Example engagement rules might be:
- Has not submitted response to this questionnaire in the last 90 days.
- Has not been asked to complete this questionnaire in the last 30 days.
- Has not been asked to complete any questionnaire in the last 7 days.

## Anatomy
<Example plain>
  <FeedbackAnatomy />
</Example>

| Label  | Region       | Purpose                                                              |                  Required                   | Notes                                                                                                      |
| ------ | ------------ | -------------------------------------------------------------------- | :-----------------------------------------: | ---------------------------------------------------------------------------------------------------------- |

> | **1**  | **Title**    | Title prompting user for their feedback.                             | <Checkmark a11yTitle="true" size="small" /> | <span>Use the [semantically correct heading level](/foundation/typography?q=semantically#semantic-usage-of-heading-levels) for the title. "We'd love your feedback" is a nice, appreciative default title.</span>       |

Reviewer 4: I do love the idea of replacing text with an icon within the table here. For consistency with the rest of our icons, should we have those be text-strong as well as our default stroke weight? These look a littler thinner.

Author: so I saw were using `checkmark` from grommet-icons for the tables and the thicker one is `FormCheckmark` so going to leave it as `checkmark` for now so its consistent across the tables but something for us to consider which one to use

> | **2**  | **Body**     | The body may be populated as many questions as appropriate.          | <Checkmark a11yTitle="true" size="small" /> | <span>Question types can include [StarRating](https://v2.grommet.io/starrating), [ThumbsRating](https://v2.grommet.io/thumbsrating), select all that apply (e.g. checkbox group), choose one (e.g. RadioButtonGroup), open ended (e.g. TextArea), or any other input types.</span> |                                                             

Reviewer 4: Change to "The body may be populated with as many questions as appropriate."

Author: changed

> | **3**  | **Footer **  | Contains two actions: "Submit" and "Dismiss" if used in Layer modal. Otherwise only "Submit" is needed inline. | <Checkmark color='text-strong' a11yTitle="true" size="small" /> |  Submit button sends survey responses to the HPE customer experience team and initiates thank you message. If presented in a modal, the Dismiss button allows the user to bypass the survey and continue their product experience. |

Reviewer 3: ```suggestion
| **3**  | **Footer **  | Contains two actions: "Submit" and "Dismiss" if used in Layer modal. Otherwise only "Submit" is needed inline. | <Checkmark color='text-strong' a11yTitle="true" size="small" /> |  Submit button sends survey responses to the HPE Customer Experience team and initiates thank you message. If presented in a modal, the Dismiss button allows the user to bypass the survey and continue their product experience. |
```

## Implementation

### Feedback question styling
Feedback surveys are composed with Grommet Form and FormFields containing any input types. Add the 
`kind="survey"` prop to Form to apply special survey styling to FormField labels and inputs.

```
<Form 
  kind="survey" // Apply feedback styling to form field labels and inputs
  ... 
>
  <FormField 
    htmlFor="question-1" 
    name="QID1" 
    label="How was the experience deploying your workload?"
  >
    <ThumbsRating id="question-1" name="QID1" />
  </FormField>
</Form>
```

### Feedback setup and submission

> To initiate user feedback collection, please contact the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>unified customer experience team</a> for the following steps:

Reviewer 3: ```suggestion
To initiate user feedback collection, please contact the <a href={`mailto:${contact.email}?subject=Collecting%20user%20feedback%20in%20a%20product%20experience`}>Unified Customer Experience team</a> for the following steps:
```

> 1. A dedicated survey will be created for your use case. At that time, an API token, survey id, and response id will be provided.

Reviewer 3: ```suggestion
1. The customer experience team will work with you to understand your needs and create a dedicated survey for your use case. At that time, an API token, survey id, and response id will be provided.
```

> 1. Obtain the survey response schema with the [retrieve a survey response call](https://api.qualtrics.com/1179a68b7183c-retrieve-a-survey-response). The properties in the "values" object represent the data which should be submitted with each new survey response.

Reviewer 3: ```suggestion
1. Map UI survey implementation to the survey's response schema. Use the API token, survey Id, and response Id to obtain the survey response schema with the [retrieve a survey response call](https://api.qualtrics.com/1179a68b7183c-retrieve-a-survey-response). The properties in the "values" object represent the data which should be submitted with each new survey response.
```

1. Survey responses are submitted using [create a new response](https://api.qualtrics.com/f1cad92018d2b-create-a-new-response).

```
// Submit survey response
fetch(`https://yul1.qualtrics.com/API/v3/surveys/${SURVEY_ID}/responses`, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'X-API-TOKEN': API_TOKEN,
  },
  body: JSON.stringify(data),
})
.then(response => {
  // Response handling here
})
.catch(error => {
  // Error handling here
});
```

## Examples

### Solicited feedback
Solicited feedback is an active method of asking users for their opinions and perspectives. 
A feedback questionnaire is presented after a user has completed a task or consumed a piece of 
content. Solicited feedback should not interrupt or impede a user from completing their intended 
objective.

#### Task flow in center layer

> Click on "Power 1 device" to see the feedback questions presented at the completion of the 

Reviewer 4: Replace with "Click on "Power on 1 device" since the button has the word "on" in it too.

Author: changed

- Any location where customer insights can help us to improve our experiences.
- To gather data to assist in advocating for changes to the user experience.
- To evaluate the effectiveness of content.

### When not to use
Don’t collect feedback in the following use cases:
- If it disrupts a user flow or is intrusive in any way.
- There is no actionable plan for what to do with the data.
- In locations where users do not have enough context to form a valid opinion. 
- There are too many implementations of user feedback collection, as not to dilute their effectiveness.

### Anatomy
<Example plain>
  <FeedbackAnatomy />
</Example>

| Label  | Region       | Purpose                                                              |                  Required                   | Notes                                                                                                      |
| ------ | ------------ | -------------------------------------------------------------------- | :-----------------------------------------: | ---------------------------------------------------------------------------------------------------------- |

> | **1**  | **Title**    | Title will inform user that their feedback or rating wou.            | <Checkmark a11yTitle="true" size="small" /> | --                                                                                                         |

Reviewer 3: ```suggestion
| **1**  | **Title**    | Title prompting user for their feedback.            | <Checkmark a11yTitle="true" size="small" /> | "We'd love your feedback" is a nice, appreciative default title.       |
```

> | **2**  | **Body**     | The body may be populated as many questions as appropriate.          | <Checkmark a11yTitle="true" size="small" /> |  Question types can include starRating, thumbsRating, select all that apply (i.e checkbox group), choose one (i.e RadioButtonGroup), open ended (i.e textarea) |                                                             

Reviewer 3: ```suggestion
| **2**  | **Body**     | The body may be populated as many questions as appropriate.          | <Checkmark a11yTitle="true" size="small" /> |  Question types can include StarRating, ThumbsRating, select all that apply (e.g. checkbox group), choose one (e.g. RadioButtonGroup), open ended (e.g. TextArea), or any other input types. |                                                             
```

> | **3**  | **Footer **  | Contains two actions: "Submit" and "Cancel"                          | <Checkmark a11yTitle="true" size="small" /> |  Submit button sends data to Qualtrics and initiates thank you message. Cancel button is optional for modal use cases. |

Reviewer 3: ```suggestion
| **3**  | **Footer **  | Contains two actions: "Submit" and "Dismiss"                          | <Checkmark a11yTitle="true" size="small" /> |  Submit button sends survey responses to the HPE customer experience team and initiates thank you message. If presented in a modal, the Dismiss button allows the user to bypass the survey and continue their product experience. |
```
